schema/city_schema.json

```diff
@@ -1,67 +1,128 @@
-							"type":"boolean"
-						},
-						"url":{
-							"description":"A URL of a web resource where more information can be viewed",
-							"type":"string"
-						},
-						"opening_hours":{
-							"description":"Opening hours of this lot in OpenStreetMap format (https://wiki.openstreetmap.org/wiki/Key:opening_hours)",
-							"type":"string"
-						},
-						"fee_hours":{
-							"description":"Hours during which usage of this lot incurrs fees in OpenStreetMap format (https://wiki.openstreetmap.org/wiki/Key:opening_hours). It is implied that outside those hours usage is free.",
-							"type":"string"
-						}
-					},
-					"required":["name", "count", "state", "forecast"]
-				}
-			}
-		},
-			"required":["name", "lots"]
-	}
-}
+  "$schema": "http://jkliemann.de/parkendd/media/city_schema.json",
+  "title": "ParkenDD JSON API",
+  "description": "JSON required by ParkenDD app to show parking lots",
+  "type": "object",
+  "properties": {
+    "last_downloaded": {
+      "type": "string",
+      "format": "date-time"
+    },
+    "last_updated": {
+      "type": "string",
+      "format": "date-time"
+    },
+    "lots": {
+      "type": "array",
+      "items": {
+        "type": "object",
+        "properties": {
+          "address": {
+            "oneOf": [
+              {
+                "type": "string"
+              },
+              {
+                "type": "null"
+              }
+            ]
+          },
+          "coords": {
+            "oneOf": [
+              {
+                "type": "object",
+                "properties": {
+                  "lat": {
+                    "type": "number",
+                    "minimum": -90,
+                    "maximum": 90
+                  },
+                  "lng": {
+                    "type": "number",
+                    "minimum": -180,
+                    "maximum": 180
+                  }
+                }
+              },
+              {
+                "type": "null"
+              }
+            ]
+          },
+          "forecast": {
+            "type": "boolean"
+          },
+          "free": {
+            "type": "integer",
+            "minimum": 0
+          },
+          "id": {
+            "type": "string"
+          },
+          "lot_type": {
+            "oneOf": [
+              {
+                "type": "string"
+              },
+              {
+                "type": "null"
+              }
+            ]
+          },
+          "name": {
+            "type": "string"
+          },
+          "region": {
+            "oneOf": [
+              {
+                "type": "string"
+              },
+              {
+                "type": "null"
+              }
+            ]
+          },
+          "state": {
+            "enum": [
+              "open",
+              "nodata",
+              "closed",
+              "many",
+              "few",
+              "full"
+            ]
+          },
```

Comment:
> Especially Aarhus and Koeln currently report states as "unknown", which IMHO should be reported as "nodata".

`unknown` and `nodata` have two dedicated meanings. `unknown` is the complement to `open` and `closed` which means that even if we have data about the number of free lots we cannot safely tell if the lot is open or closed (some cities do not provide this information). In this case the apps would still show it as open since this is usually the case. `nodata` means we do not have data about the number of free lots.

What are the definitions of `many`, `few` and `full`? Is this only used if there is no more detailed data available? Otherwise I'd say this information is included in the number of free and total lots. Another problem is that these states are not supported by any app yet (although this might be one that can be solved without too much effort).

Reply:
We implemented these states for Herrrenberg, but I'm fine with using state only for open/closed or unkown, if no information is known. Isn't `nodata` redundant for having no count information and additionally hide open/close then?

Reply:
> Isn't `nodata` redundant for having no count information and additionally hide open/close then?

I'm not sure if I understand the question correctly. `nodata` implies that there is no live data or not enough to be useful. It implies `unknown`. The desired handling for applications is that if `nodata` is set the application simply shows the parking lot but only with static data (location, type, total count, etc.). The case that we know if a lot is open or closed but we don't have data on how full it is cannot be modeled currently.

Reply:
Is it really necessary to have a state `nodata`? Couldn't this implicitly deduced when no `free` value is given? Currently, there are instances, which return free=-1 if no data is available. Separating opening state and realtime availability in two different property allows to communicate open/closed state without knowing current occupancy.

My proposition is to keep only open/closed/unknown and remove nodata as well as our introduced few/many/full values.

Reply:
We have to distinguish between what is logically useful (where you are completely correct) and what we need for compatibility. My current motivation is to keep the latter (where we also need `nodata`). We are aware that the current API has issues but we don't want to "fix" it in a way that would break compatibility since it has some drawbacks that cannot be fixed at all with this architecture. We have outlined the main problems with the current API in https://github.com/offenesdresden/ParkAPI/issues/130.
Further I know that removing `nodata` will break some of our applications. Also I have noticed that there are a few third party implementations that use our API where I don't know how exactly they are implemented and I don't want to break them either with such a small change.

Reply:
I'm fine with leaving `nodata` in the schema for know, though I think that clients should be capable to handle a missing free value as well. I'll adapt this PR (removing the frozen feed parser version as well) after #222 has been fixed.

Reply:
Now fixed